### PR TITLE
`azurerm_mssql_database` support 0 for  `min_capacity`

### LINKED ISSUE #8226: azurerm_mssql_elasticpool - service tier 'Basic' must have whole numbers as their 'minCapacity'
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform v0.13.0
+ provider registry.terraform.io/hashicorp/azurerm v2.24.0

### Affected Resource(s)

* `azurerm_mssql_elasticpool`

### Terraform Configuration Files

```hcl
resource "azurerm_mssql_elasticpool" "sql_elastic_pool" {
  name = "agency-dept-dev-app-epool"
  location = "eastus"
  resource_group_name = azurerm_resource_group.resource_group.name
  server_name = azurerm_mssql_server.sql_server.name
  max_size_gb = 4.8828125


  per_database_settings {
    min_capacity = 0.5
    max_capacity = 5
  }

  sku {
    capacity = 50
    name = "BasicPool"
    tier = "Basic"
  }
  depends_on = [azurerm_mssql_server.sql_server]
}
```

### Expected Behavior

a sql elastic pool is created

### Actual Behavior
```
Error: service tier 'Basic' must have whole numbers as their 'minCapacity'

  on main.tf line 46, in resource "azurerm_mssql_elasticpool" "sql_elastic_pool":
  46: resource "azurerm_mssql_elasticpool" "sql_elastic_pool" {
```
### Steps to Reproduce

1. `terraform plan`

### References
According to [this link](https://docs.microsoft.com/en-us/azure/azure-sql/database/resource-limits-dtu-elastic-pools) , basic elastic pools have a Min DTU per database choices valid option of only 0,.5. When I attempt to set the value to 0.5, I receive the error above. When I set it to a whole number, ie: 1, I get 
```
Error: sql.ElasticPoolsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidInputValueForDatabaseDtuMin" Message="The DTU min per database (1) for the elastic pool does not belong to the allowed values for service tier 'Basic'."

  on dfr-impact-env.tf line 240, in resource "azurerm_mssql_elasticpool" "sql_elastic_pool":
 240: resource "azurerm_mssql_elasticpool" "sql_elastic_pool" {
```

This fails both in Azure DevOps Pipelines and manually with terraform plan.

When I create a database within the portal, since I can't do it with terraform, the min_capacity is set to 0. Which also doesn't appear to be valid but yet, that's how they are created and I see no way to change that during portal creation. I can't seem to create or import a mssql database using terraform. This is kind of a big deal.

* #0000

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/8226

### CODE REVIEW

File: azurerm/internal/services/mssql/mssql_database_resource.go [hunk 1]
```diff
@@ -123,7 +122,7 @@ func resourceArmMsSqlDatabase() *schema.Resource {
 				Type:         schema.TypeFloat,
 				Optional:     true,
 				Computed:     true,
-				ValidateFunc: azValidate.FloatInSlice([]float64{0.5, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.25, 2.5, 3, 4, 5}),
+				ValidateFunc: validation.FloatAtLeast(0),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/mssql/mssql_database_resource.go:125]: alternatively we could add `0` as an allowed value here:

```suggestion
				ValidateFunc: azValidate.FloatInSlice([]float64{0, 0.5, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.25, 2.5, 3, 4, 5}),
```

can we also add an acceptance test for this SKU?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @yupwei68 

Thanks for this PR.

Taking a look through here it appears that we can add `0` as an allowed value here (as `1` and `2` are) - which should solve this and provide examples of valid values here; at the same time can we also add an acceptance test to cover this missing use-case? If we can fix those up then we should be able to take another look

Thanks!

**Comment by yupwei68** [comment]
Hi @tombuildsstuff Thanks for your comments. Please continue reviewing.

File: azurerm/internal/services/mssql/mssql_database_resource.go [hunk 2]
```diff
@@ -123,7 +122,7 @@ func resourceArmMsSqlDatabase() *schema.Resource {
 				Type:         schema.TypeFloat,
 				Optional:     true,
 				Computed:     true,
-				ValidateFunc: azValidate.FloatInSlice([]float64{0.5, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.25, 2.5, 3, 4, 5}),
+				ValidateFunc: validate.MsSqlDatabaseMinCapacity,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/mssql/mssql_database_resource.go:125]: can we add an [Acceptance Test](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/mssql/tests/mssql_database_resource_test.go#L16-L33) to provision a Database with the Capacity of `0` to confirm the Basic SKU / #8226 is fixed

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @yupwei68 

Taking a look through the unit test looks good, but can we add an acceptance test to ensure that #8226 is fixed?

Thanks!

**Pull request merged**